### PR TITLE
Advance Game Port (AGP) fixes

### PR DESCRIPTION
**Pull request by skidau**
Added a generic hash key finder.
Fixed the memcard and AGP path parsing in Windows (the code is expecting '/' to be used as the DIR_SEP).

### CODE REVIEW

**Comment by JMC47** [comment]
Fixes the Advance Game Port on Windows, at least for portable mode.

File: Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ void EmuCodeBlock::SetFPRF(Gen::X64Reg xmm)
 void EmuCodeBlock::JitGetAndClearCAOV(bool oe)
 {
 	if (oe)
-		AND(8, PPCSTATE(xer_so_ov), Imm32(~XER_OV_MASK)); //XER.OV = 0
+		AND(8, PPCSTATE(xer_so_ov), Imm8(~XER_OV_MASK)); //XER.OV = 0
```
Comment (magumagu) [inline Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp]: This seems unrelated.

File: Source/Core/Core/HW/EXI_DeviceAGP.cpp [hunk 2]
```diff
@@ -111,11 +111,21 @@ void CEXIAgp::LoadHash()
...
+		{
+			if (Memory::ReadUnchecked_U32(h) == 0x005ebce2)
+				hash_addr = h;
+		}
+
 		for (int i = 0; i < 0x100; i++)
 		{
 			// Load the ROM hash expected by the AGP
-			m_hash_array[i] = Memory::ReadUnchecked_U8(0x0017e908 + i);
+			m_hash_array[i] = Memory::ReadUnchecked_U8(hash_addr + i);
```
Comment (magumagu) [inline Source/Core/Core/HW/EXI_DeviceAGP.cpp]: Is this a number we can compute somehow?  Or is it just a fixed 256 bytes we could hardcode into Dolphin?  I mean, we don't want to cause mysterious errors if someone ever tries to write AGP homebrew.

Comment (BhaaLseN) [inline Source/Core/Core/HW/EXI_DeviceAGP.cpp]: If we cannot compute it, make it a define/constant/whatever, and/or at least comment where the numbers comes from and what it does.

File: Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp (hunk 1 above)
Comment (JMC47) [inline Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp]: AGP runs into a CPU bug here.  It's unrelated to the AGP, but needs to be stuffed into a PR somewhere.

Comment (magumagu) [inline Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp]: Okay.  I'd like to see some hwtests to make sure this actually works correctly.

File: Source/Core/Core/HW/EXI_DeviceAGP.cpp (hunk 2 above)
Comment (tueidj) [inline Source/Core/Core/HW/EXI_DeviceAGP.cpp]: Yep, this code is dumb; it looks for a hardcoded value to determine where to find the hash values (and also depends on the final element being 0x35) in preference to just using predefined values?
The hash array is baked into the hardware, just declare it const and fill it with the expected values.

**Comment by tueidj** [comment]
That hash is crc8.

**Comment by skidau** [comment]
Thanks for the tip, @tueidj

File: Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp (hunk 1 above)
Comment (FioraAeterna) [inline Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp]: Yeah, certain no guarantee our overflow handling is 100% correct even for the parts we do implement, but that was just a dumb typo in an older patch of mine months ago which never triggered because nothing used that code.
